### PR TITLE
Remove the Android API level enforcement.

### LINKED ISSUE #744: Change min java/android version for SDK
The proposal is to increase the minimum java version and android api level only for the SDK (the API will remain the same, in order to allow gRPC to depend on opentelemetry-api).

The new proposed versions are (some details here https://developer.android.com/studio/write/java8-support):
* Java 8
* Android 24

I would like every vendor to comment on this.

### PR DESCRIPTION
**Pull request by jkwatson**
Keep the Java 7 enforcement.
Resolves #744

### CODE REVIEW

**Comment by jkwatson** [comment]
@carlosalberto re-opened without the proto submodule messup

**Comment by Oberon00** [comment]
Shouldn't we enforce another (higher) android API level instead? There might still be some things in Java 7 (like AWT) that are not Android-compatible.

**Comment by jkwatson** [comment]
I'm of the opinion that the default SDK is unlikely to be optimal to run on android, since the characteristics of the operating environment are very different than a standard JVM. I'd be ok with enforcing android 24 for the API, but I don't think there's a lot of value in enforcing android for the default SDK.

File: context_prop/build.gradle [hunk 1]
```diff
@@ -5,7 +5,6 @@ dependencies {
     api libraries.grpc_context
 
     signature "org.codehaus.mojo.signature:java17:1.0@signature"
-    signature "net.sf.androidscents.signature:android-api-level-14:4.0_r4@signature"
```
Comment (bogdandrutu) [inline context_prop/build.gradle]: this needs to be API-14, api depends on this.

File: sdk/build.gradle [hunk 2]
```diff
@@ -9,7 +9,6 @@ dependencies {
     annotationProcessor libraries.auto_value
 
     signature "org.codehaus.mojo.signature:java17:1.0@signature"
-    signature "net.sf.androidscents.signature:android-api-level-14:4.0_r4@signature"
```
Comment (bogdandrutu) [inline sdk/build.gradle]: maybe keep here api24?

File: context_prop/build.gradle (hunk 1 above)
Comment (jkwatson) [reply context_prop/build.gradle]: done

File: sdk/build.gradle (hunk 2 above)
Comment (jkwatson) [reply sdk/build.gradle]: 👍

**Comment by jkwatson** [comment]
ok, I got everything tidied up and ship-shape, I think.

**Pull request merged**